heat/core/linalg/basics.py

```diff
@@ -40,6 +41,116 @@
 ]
 
 
+def det(a: DNDarray) -> DNDarray:
+    """
+    Returns the determinant of a square matrix.
+
+    Parameters
+    ----------
+    a : DNDarray
+        A square matrix or a stack of matrices. Shape = (...,M,M)
+
+    Raises
+    ------
+    TypeError
+        If the dtype of 'a' is not floating-point.
+    ValueError
+        If `a.ndim < 2` or if the length of the last two dimensions is not the same.
+
+    Examples
+    --------
+    >>> a = ht.array([[-2,-1,2],[2,1,4],[-3,3,-1]])
+    >>> ht.linalg.det(a)
+    DNDarray(54., dtype=ht.float64, device=cpu:0, split=None)
+    """
+    # no split in the square matrices
+    if a.split is None or a.split < a.ndim - 2:
```

Comment:
```suggestion
    if not a.is_distributed() or a.split < a.ndim - 2:
```
might need to change the error messages to be consistent